Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -527,7 +548,24 @@ extension SourceKitServer {
       executeCommandProvider: ExecuteCommandOptions(
         commands: builtinSwiftCommands // FIXME: Clangd commands?
       )
-    )))
+    )
+  }
+
+  func registerCapabilities(
+    for server: ServerCapabilities,
+    languages: [Language],
+    registry: CapabilityRegistry
+  ) {
+    guard let completionOptions = server.completionProvider else { return }
+    guard registry.completionSupported && !registry.hasCompletionRegistrations(for: languages) else { return }
```

Comment:
Why does it bail out if there are already registrations? I would have expected we'd check if they're equal.

Reply:
If we have a conflicting registration for completion there's not much we can do unless we try to manually merge them - I guess I could just log it if that happens?

Reply:
Yeah, how about we log a warning here unless the options are identical to what's already registered for that language.